itest/mint_fund_seal_test.go

```diff
@@ -0,0 +1,706 @@
+	"github.com/lightninglabs/taproot-assets/taprpc/mintrpc"
+	"github.com/lightninglabs/taproot-assets/taprpc/universerpc"
+	"github.com/lightninglabs/taproot-assets/tapsend"
+	"github.com/lightninglabs/taproot-assets/universe"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/stretchr/testify/require"
+)
+
+// testMintFundSealAssets tests that we're able to mint assets with custom
+// script keys, group keys, and group key tapscript roots. Once minted, we
+// also want to reissue assets with a custom asset group witness, and spend
+// an asset with a custom script key witness. An asset group with a tapscript
+// root should also be imported into a new node correctly.
+func testMintFundSealAssets(t *harnessTest) {
+	// We create a second tapd node that will be used to simulate a second
+	// party in the test. This tapd node is connected to lnd "Bob".
+	var (
+		aliceTapd = t.tapd
+		aliceLnd  = t.lndHarness.Alice
+		bobLnd    = t.lndHarness.Bob
+		bobTapd   = setupTapdHarness(t.t, t, bobLnd, t.universeServer)
+	)
+
+	aliceLndClient, err := t.newLndClient(aliceLnd)
+	require.NoError(t.t, err)
+
+	defer func() {
+		require.NoError(t.t, bobTapd.stop(!*noDelete))
+	}()
+
+	displayGroupWitness := func(wit *taprpc.GroupWitness) string {
+		return fmt.Sprintf(
+			"genID: %x\nWitness: %x\n",
+			wit.GenesisId, wit.Witness,
+		)
+	}
+
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultWaitTimeout)
+	defer cancel()
+
+	aliceLndKeyRing := taprootassets.NewLndRpcKeyRing(
+		&aliceLndClient.LndServices,
+	)
+
+	// Derive a random key on an LND node, with a key family not matching
+	// the Taproot Assets key family.
+	deriveRandomKey := func(
+		keyRing *taprootassets.LndRpcKeyRing) keychain.KeyDescriptor {
+
+		var (
+			randFam = test.RandInt31n(math.MaxInt32)
+			randInd = test.RandInt31n(255)
+			desc    keychain.KeyDescriptor
+			err     error
+		)
+
+		// Ensure that we use a different key family from tapd.
+		for randFam == asset.TaprootAssetsKeyFamily {
+			randFam = test.RandInt31n(math.MaxInt32)
+		}
+
+		// Use DeriveNextKey to reach our desired key index.
+		for i := int32(0); i != randInd; i++ {
+			desc, err = keyRing.DeriveNextKey(
+				ctxt, keychain.KeyFamily(randFam),
+			)
+			require.NoError(t.t, err)
+		}
+
+		return desc
+	}
+	_ = deriveRandomKey(aliceLndKeyRing)
+
+	// Let's derive the keys and tapscript trees we'll use.
+	// tweakedScript will have an internal key not managed by a tapd, and
+	// a tapscript root with a hashlock and a single sig script.
+	tweakedScriptDesc := deriveRandomKey(aliceLndKeyRing)
+	tweakedScriptHashLock := test.ScriptHashLock(
+		t.t, bytes.Clone(test.DefaultHashLockWitness),
+	)
+	tweakedScriptSigLock := test.ScriptSchnorrSig(
+		t.t, tweakedScriptDesc.PubKey,
+	)
+	tweakedScriptTapTree := txscript.AssembleTaprootScriptTree(
+		tweakedScriptHashLock, tweakedScriptSigLock,
+	)
+	tweakedScriptTweak := txscript.NewTapBranch(
```

Comment:
nit: this could just be `tweakedScriptTapTree.Root.TapHash()`, no need to create a branch. Same further below.